pystac_client/client.py

```diff
@@ -132,6 +132,8 @@ def open(
         Return:
             catalog : A :class:`Client` instance for this Catalog/API
         """
+        if url.endswith("/"):
+            url = url[0:-1]
```

Comment:
Just a style suggestion.

```suggestion
url = url.rstrip("/")
```

I guess technically mine would change a URL like `/v1//` to `/v1`. Do we have a preference?